fix(treesitter): disable injections by commenting out queries (#31748)
neovim/neovim: runtime/doc/treesitter.txt

    • `injection.parent` - indicates that the captured node’s text should
      be parsed with the same language as the node's parent LanguageTree.

Injection queries are currently run over the entire buffer, which can be slow
for large buffers. To disable injections for a certain language, just place an
empty query file in your 'runtimepath'. For example, to disable injections for
`c`, use the following `queries/c/injections.scm`: >query

>     ; disabled

Reviewer: This doesn't do anything right? Just the existence of the file disables the injections? Unlike `; inherits` and `; extends`.

If so then this is slightly confusing.

Author: Yeah, true. Should I just remove this example then? Maybe does more harm than good.

Reviewer 2: I think it's better to explicitly describe what happens, and/or use a comment that us obviously arbitrary.

    • `injection.parent` - indicates that the captured node’s text should
      be parsed with the same language as the node's parent LanguageTree.

Injection queries are currently run over the entire buffer, which can be slow
for large buffers. To disable injections for a certain language, just place an

> empty query file in your 'runtimepath'. For example, to disable injections for

Reviewer 2: ```suggestion
query file containing only a comment in your 'runtimepath'. For example, to disable injections for
```

("empty" would mean "no contents", not even comments)

Reviewer: That would still work.

Reviewer 2: You mean adding an empty file? I was under the impression that it had to have _some_ contents and empty file was not enough. In that case, the documentation can (and should) be simplified significantly.

Reviewer: The file just needs to exist to prevent other files being loaded. We can then use `extends` or `inherits` to pull in other files.

Reviewer 2: OK, then that was a misunderstanding on my part; let's just document empty query then.

`c`, use the following `queries/c/injections.scm`: >query

>     ; This file contains no captures (only comments), thus injections are
>     ; ignored for this language.
>     ; Note that these comments are arbitrary and not necessary.

Reviewer 2: ```suggestion
    ; This file intentionally left blank.
```

> Injection queries are currently run over the entire buffer, which can be slow
> for large buffers. To disable injections for a certain language, just place an
> empty query file in your 'runtimepath'. For example, to disable injections for
> `c`, use the following `queries/c/injections.scm`: >query
> 
>     ; This file contains no captures (only comments), thus injections are
>     ; ignored for this language.
>     ; Note that these comments are arbitrary and not necessary.
> 

Reviewer 2: ```suggestion
Injection queries are currently run over the entire buffer, which can be slow
for large buffers. To disable injections for, e.g.,  `c`, just place an
empty `queries/c/injections.scm` file in your 'runtimepath'.
```

Author: But wouldn't it also be helpful to know that the file can be either empty, or contain only comments?

Reviewer 2: At this point, people only need to know how to disable injections, and the simplest way will do. (It just has to be correct, not complete.)

Reviewer 2: (and I think if the comment happens to be `; extends`, then it _won't_ disable anything -- and now things become too complicated to explain in full)

Author: Sure. I still feel like I would personally get confused by an empty file and think it was a mistake, thus I would like  a comment explaining its purpose, but I guess the user could fiddle around for 30 seconds and figure things out
